flink-runtime/src/test/java/org/apache/flink/runtime/rest/handler/legacy/backpressure/BackPressureRequestCoordinatorTest.java

```diff
@@ -0,0 +1,334 @@
+	private BackPressureRequestCoordinator coordinator;
+
+	@BeforeClass
+	public static void setUp() throws Exception {
+		executorService = new ScheduledThreadPoolExecutor(1);
+	}
+
+	@AfterClass
+	public static void tearDown() throws Exception {
+		if (executorService != null) {
+			executorService.shutdown();
+		}
+	}
+
+	@Before
+	public void initCoordinator() throws Exception {
+		coordinator = new BackPressureRequestCoordinator(executorService, requestTimeout);
+	}
+
+	@After
+	public void shutdownCoordinator() throws Exception {
+		if (coordinator != null) {
+			coordinator.shutDown();
+		}
+	}
+
+	/** Tests simple request of task back pressure stats. */
+	@Test(timeout = 10000L)
+	public void testSuccessfulBackPressureRequest() throws Exception {
+		ExecutionVertex[] vertices = createExecutionVertices(ExecutionState.RUNNING, CompletionType.SUCCESSFULLY);
+
+		CompletableFuture<BackPressureStats> requestFuture = coordinator.triggerBackPressureRequest(vertices);
+		BackPressureStats backPressureStats = requestFuture.get();
+
+		// verify the request result
+		assertEquals(0, backPressureStats.getRequestId());
+		Map<ExecutionAttemptID, Double> backPressureRatios = backPressureStats.getBackPressureRatios();
+		for (ExecutionVertex executionVertex : vertices) {
+			ExecutionAttemptID executionId = executionVertex.getCurrentExecutionAttempt().getAttemptId();
+			assertEquals(backPressureRatio, backPressureRatios.get(executionId), 0.0);
+		}
+
+		// verify no more pending request
+		assertEquals(0, coordinator.getNumberOfPendingRequests());
+	}
+
+	/** Tests back pressure request of non-running tasks fails the future. */
+	@Test
+	public void testRequestNotRunningTasks() throws Exception {
+		ExecutionVertex[] vertices = createExecutionVertices(ExecutionState.DEPLOYING, CompletionType.SUCCESSFULLY);
+
+		CompletableFuture<BackPressureStats> requestFuture = coordinator.triggerBackPressureRequest(vertices);
+		assertTrue(requestFuture.isDone());
+		try {
+			requestFuture.get();
+			fail("Exception expected.");
+		} catch (ExecutionException e) {
+			assertTrue(e.getCause() instanceof IllegalStateException);
+		}
+	}
+
+	/** Tests failed request to execution fails the future. */
+	@Test(timeout = 10000L)
+	public void testBackPressureRequestWithException() throws Exception {
+		ExecutionVertex[] vertices = createExecutionVertices(ExecutionState.RUNNING, CompletionType.EXCEPTIONALLY);
+
+		CompletableFuture<BackPressureStats> requestFuture = coordinator.triggerBackPressureRequest(vertices);
+		try {
+			requestFuture.get();
+			fail("Exception expected.");
+		} catch (ExecutionException e) {
+			assertTrue(e.getCause() instanceof RuntimeException);
+		}
+	}
+
+	/** Tests that request timeout if not finished in time. */
+	@Test(timeout = 10000L)
+	public void testBackPressureRequestTimeout() throws Exception {
+		long requestTimeout = 1000;
+		coordinator = new BackPressureRequestCoordinator(executorService, requestTimeout);
```

Comment:
Do not reuse this `coordinator`, it might cause the previous created `coordinator` leak. We can define a new variable here and shutdown it in finally.